onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.cc

```diff
@@ -575,31 +551,76 @@ Status ModelBuilder::Compile(std::unique_ptr<Model>& model) {
       // and cpu is not in the target devices list
       return ORT_MAKE_STATUS(ONNXRUNTIME, EP_FAIL,
                              "The model cannot run using current set of target devices, ",
-                             nnapi_target_devices_detail_);
-    } else {
+                             GetDevicesDescription(nnapi_target_devices_));
+    // workaround for bugs in Android OS. sometimes ops are passed checking but failed at compilation.
+    }
+    // else // no else after return
+    if (target_device_option_ != TargetDeviceOption::ALL_DEVICES) {
       use_create_for_devices = true;
     }
   }
 
+#ifndef NDEBUG
+  if (nnapi_target_devices_.size() > 1 && nnapi_target_devices_.back().type == ANEURALNETWORKS_DEVICE_CPU) {
+    auto supported_ops = gsl::make_span(supported_ops_holder.get(), num_nnapi_ops_);
+    RETURN_STATUS_ON_ERROR_WITH_NOTE(
+        nnapi_.ANeuralNetworksModel_getSupportedOperationsForDevices(
+            nnapi_model_->model_, device_handles.data(),
+            static_cast<uint32_t>(device_handles.size() - 1), supported_ops.data()),
+        "on getSupportedOperationsForDevices");
+
+    ORT_ENFORCE(num_nnapi_ops_==operations_recorder_.size(), "num_nnapi_ops_!=operations_recorder_.size()");
+    std::unordered_map<std::string, std::pair<int32_t, int32_t>> optype_support_status;
+    for (size_t idx = 0; idx < operations_recorder_.size(); idx++) {
+      auto [onnx_node_idx, nnapi_idx] = operations_recorder_[idx];
+      const auto* node(graph_viewer_.GetNode(onnx_node_idx));
+      auto stat_name = node->OpType() + ".nnapi_op_" + std::to_string(nnapi_idx);
+
+      if (!supported_ops[idx]) {
+        optype_support_status[stat_name].first++;
+      } else {
+        optype_support_status[stat_name].second++;
+      }
+    }
+    size_t total_ops = 0;
+    std::string fallback_op_detail, normal_op_detail;
+
+    for (const auto& [op, ops_status] : optype_support_status) {
+      auto& [support_cnt, unspport_cnt] = ops_status;
```

Comment:
is the unsupported count first? actually, the rest of the loop looks ok. maybe the variable names should just be swapped.

Reply:
Thanks. Resolved by removing the Not operator above.